### PR TITLE
b2b-shipment-smole-test

### PR DESCRIPTION
**Pull request by durgesh-verma-spryker**
https://spryker.atlassian.net/browse/CC-21651

Orders
-Create shipment
-Edit shipment

### CODE REVIEW

File: resources/steps/zed_order_shipment_steps.robot [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    Click    ${edit_shipment_button} 
+    Select From List By Label    ${delivery_address_edit_shipment_page}    New address
+    Zed: new shipment for orders details:
+    ...    || fname                             | lname                    |    email                                 |    address1              |    address2              |    city        |    zipcode     ||
+    ...    || ${yves_user_first_name}${random}  | ${yves_user_last_name}   |    ${yves_user_email}${random}           |    address1+${random}    |    address2+${random}    |    Berlin      |    11${random} ||
+    Input Text    ${shipment_address_address1}    aaa11
+    Select From List By Text    ${shipment_method_dropdown}    DHL - Express
+    Wait Until Element Is Visible    ${save_button_edit_shipment_page}
+    Click    ${save_button_edit_shipment_page}
+Zed: create shipment
```
Comment (OKrivtsova) [inline resources/steps/zed_order_shipment_steps.robot:20]: rename to Zed: Create a new shipment for order

File: resources/steps/zed_order_shipment_steps.robot [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    Zed: new shipment for orders details:
+    ...    || fname                             | lname                    |    email                                 |    address1              |    address2              |    city        |    zipcode     ||
+    ...    || ${yves_user_first_name}${random}  | ${yves_user_last_name}   |    ${yves_user_email}${random}           |    address1+${random}    |    address2+${random}    |    Berlin      |    11${random} ||
+    Input Text    ${shipment_address_address1}    aaa11
+    Select From List By Text    ${shipment_method_dropdown}    DHL - Express
+    Wait Until Element Is Visible    ${save_button_edit_shipment_page}
+    Click    ${save_button_edit_shipment_page}
+Zed: create shipment 
+    Hover    ${view_order_last_placed} 
+    Click    ${view_order_last_placed}
```
Comment (OKrivtsova) [inline resources/steps/zed_order_shipment_steps.robot:22]: this should be handled outside of your keyword using common_zed.robot keywords

File: resources/steps/zed_order_shipment_steps.robot [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        IF    '${key}'=='lname'    Type Text    ${shipment_address_last_name}    ${value}
+        IF    '${key}'=='email'    Type Text    ${shipment_address_email}    ${value}
+        IF    '${key}'=='Address1'    Type Text    ${shipment_address_address1}     ${value}
+        IF    '${key}'=='address2'    Type Text    ${shipment_address_address2}     ${value}
+        IF    '${key}'=='city'    Type Text    ${shipment_address_city}     ${value}
+        IF    '${key}'=='zipcode'    Type Text    ${shipment_address_zipcode}     ${value}
+    END
+    Select From List By Value    ${salutation_create_new_shipment_page}    Mr
+    Select From List By Index    ${country_create_new_shipment_page}    4    
+    Select From List By Text    ${shipment_method_dropdown_new_shipment_page}    DHL - Express
```
Comment (OKrivtsova) [inline resources/steps/zed_order_shipment_steps.robot:45]: why don't we send these values as parameters?

File: resources/steps/zed_order_shipment_steps.robot [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    Hover    ${view_order_last_placed} 
+    Click    ${view_order_last_placed} 
+    Click    ${create_shipment_button}
+    Zed: new shipment for orders details:
+    ...    || fname                             | lname                    |    email                                 |    address1              |    address2              |    city        |    zipcode     ||
+    ...    || ${yves_user_first_name}${random}  | ${yves_user_last_name}   |    ${yves_user_email}${random}           |    address1+${random}    |    address2+${random}    |    Berlin      |    11${random} ||
+    Input Text    ${shipment_address_address1}    aaa11
+    Wait Until Element Is Visible    ${save_button_new_shipment_page}
+    Click    ${save_button_new_shipment_page}
+Zed: new shipment for orders details:
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/steps/zed_order_shipment_steps.robot:30]: Done

File: resources/steps/zed_order_shipment_steps.robot [hunk 5]
```diff
@@ -0,0 +1,45 @@
+*** Settings ***
+Resource    ../common/common_zed.robot
+Resource    ../common/common.robot
+Resource    ../../resources/common/common_yves.robot
+Resource    ../../resources/pages/zed/zed_shipment_page.robot
+
+*** Keywords ***
+Zed: edit shipment
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/steps/zed_order_shipment_steps.robot:8]: Done

File: resources/pages/zed/zed_shipment_page.robot [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+${delivery_address_edit_shipment_page}    id=shipment_group_form_shipment_shippingAddress_idCustomerAddress
+${salutation_create_new_shipment_page}    xpath=//select[@id='shipment_group_form_shipment_shippingAddress_salutation']
+${country_create_new_shipment_page}    xpath=(//select[@id='shipment_group_form_shipment_shippingAddress_iso2Code'])[1]
+${save_button_new_shipment_page}    xpath=//input[@value='Save']
+${shipment_method_dropdown_new_shipment_page}    xpath=//select[@id='shipment_group_form_shipment_method_idShipmentMethod']
+${shipment_address_first_name}    xpath=//input[@id='shipment_group_form_shipment_shippingAddress_firstName']
+${shipment_address_last_name}    xpath=//input[@id='shipment_group_form_shipment_shippingAddress_lastName']
+${shipment_address_email}    xpath=//input[@id='shipment_group_form_shipment_shippingAddress_email']
+${shipment_address_address1}    xpath= (//input[@id="shipment_group_form_shipment_shippingAddress_address1"])[1]
+${shipment_address_address2}    xpath=(//input[@id='shipment_group_form_shipment_shippingAddress_address2'])[1]
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/pages/zed/zed_shipment_page.robot:16]: Done

File: resources/pages/zed/zed_shipment_page.robot [hunk 7]
```diff
@@ -0,0 +1,18 @@
+*** Variables ***
+${create_shipment_button}    //a[normalize-space()='Create Shipment']
+${edit_shipment_button}    (//a[contains(text(),'Edit Shipment')])[1]
+${shipment_method_dropdown}    id=shipment_group_form_shipment_method_idShipmentMethod
+${save_button_edit_shipment_page}    xpath=//input[@value='Save']
+${view_order_last_placed}    xpath=(//a[starts-with(@href,'/sales/detail')])[1]
+${delivery_address_edit_shipment_page}    id=shipment_group_form_shipment_shippingAddress_idCustomerAddress
+${salutation_create_new_shipment_page}    xpath=//select[@id='shipment_group_form_shipment_shippingAddress_salutation']
+${country_create_new_shipment_page}    xpath=(//select[@id='shipment_group_form_shipment_shippingAddress_iso2Code'])[1]
+${save_button_new_shipment_page}    xpath=//input[@value='Save']
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/pages/zed/zed_shipment_page.robot:10]: Done

File: resources/pages/zed/zed_shipment_page.robot [hunk 8]
```diff
@@ -0,0 +1,18 @@
+*** Variables ***
+${create_shipment_button}    //a[normalize-space()='Create Shipment']
+${edit_shipment_button}    (//a[contains(text(),'Edit Shipment')])[1]
+${shipment_method_dropdown}    id=shipment_group_form_shipment_method_idShipmentMethod
+${save_button_edit_shipment_page}    xpath=//input[@value='Save']
+${view_order_last_placed}    xpath=(//a[starts-with(@href,'/sales/detail')])[1]
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/pages/zed/zed_shipment_page.robot:6]: Done

File: resources/pages/zed/zed_shipment_page.robot [hunk 9]
```diff
@@ -0,0 +1,18 @@
+*** Variables ***
+${create_shipment_button}    //a[normalize-space()='Create Shipment']
+${edit_shipment_button}    (//a[contains(text(),'Edit Shipment')])[1]
+${shipment_method_dropdown}    id=shipment_group_form_shipment_method_idShipmentMethod
+${save_button_edit_shipment_page}    xpath=//input[@value='Save']
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/pages/zed/zed_shipment_page.robot:5]: Done

File: resources/pages/zed/zed_shipment_page.robot [hunk 10]
```diff
@@ -0,0 +1,18 @@
+*** Variables ***
+${create_shipment_button}    //a[normalize-space()='Create Shipment']
+${edit_shipment_button}    (//a[contains(text(),'Edit Shipment')])[1]
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/pages/zed/zed_shipment_page.robot:3]: Done

File: resources/pages/zed/zed_shipment_page.robot [hunk 11]
```diff
@@ -0,0 +1,18 @@
+*** Variables ***
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/pages/zed/zed_shipment_page.robot:1]: Done

File: resources/steps/zed_order_shipment_steps.robot [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+    Hover    ${view_order_last_placed} 
+    Click    ${view_order_last_placed} 
+    Click    ${edit_shipment_button} 
+    Select From List By Label    ${delivery_address_edit_shipment_page}    New address
+    Zed: new shipment for orders details:
+    ...    || fname                             | lname                    |    email                                 |    address1              |    address2              |    city        |    zipcode     ||
+    ...    || ${yves_user_first_name}${random}  | ${yves_user_last_name}   |    ${yves_user_email}${random}           |    address1+${random}    |    address2+${random}    |    Berlin      |    11${random} ||
+    Input Text    ${shipment_address_address1}    aaa11
+    Select From List By Text    ${shipment_method_dropdown}    DHL - Express
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/steps/zed_order_shipment_steps.robot:17]: Done

File: resources/steps/zed_order_shipment_steps.robot [hunk 13]
```diff
@@ -0,0 +1,45 @@
+*** Settings ***
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/steps/zed_order_shipment_steps.robot:1]: Done

File: tests/smoke/smoke_b2b.robot [hunk 14]
```diff
@@ -866,4 +867,43 @@ Refunds
...
+    Yves: billing address same as shipping address:    true
+    Yves: select the following existing address on the checkout as 'shipping' address and go next:    ${yves_company_user_shared_permission_owner_address}
+    Yves: select the following shipping method on the checkout and go next:    Express
+    Yves: select the following payment method on the checkout and go next:    Invoice
+    Yves: accept the terms and conditions:    true
+    Yves: 'submit the order' on the summary page
+    Yves: 'Thank you' page is displayed
+    [Teardown]    Run Keywords    Yves: logout on Yves as a customer
+    ...    AND    Reload
+Creating_shipment_in_zed
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot:898]: Done

File: resources/steps/zed_order_shipment_steps.robot [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+Resource    ../common/common_zed.robot
+Resource    ../common/common.robot
+Resource    ../../resources/common/common_yves.robot
+Resource    ../../resources/pages/zed/zed_shipment_page.robot
+
+*** Keywords ***
+Zed: edit shipment
+    Hover    ${view_order_last_placed} 
+    Click    ${view_order_last_placed} 
+    Click    ${edit_shipment_button}
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/steps/zed_order_shipment_steps.robot:11]: Done

File: resources/steps/zed_order_shipment_steps.robot (hunk 2 above)
Comment (durgesh-verma-spryker) [reply resources/steps/zed_order_shipment_steps.robot:22]: Done

File: resources/steps/zed_order_shipment_steps.robot (hunk 3 above)
Comment (durgesh-verma-spryker) [reply resources/steps/zed_order_shipment_steps.robot:45]: Done

File: resources/steps/zed_order_shipment_steps.robot (hunk 13 above)
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/steps/zed_order_shipment_steps.robot:1]: Done

**Review by OKrivtsova — APPROVED** [review]

**Comment by durgesh-verma-spryker** [comment]
resolved conflict

**Review by OKrivtsova — APPROVED** [review]

**Pull request merged**